### PR TITLE
Task: optimize update Forecast should run twice.

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
Updating the state to allow for two forecast runs. 
This will be later utilized in the output graphs.
In the meantime everything should continue to use the post optimized forecast to prevent functional changes.

Also add on-focus-out to tera-input

### CODE REVIEW

File: packages/client/hmi-client/src/components/widgets/tera-input.vue [hunk 1]
```diff
@@ -28,7 +29,7 @@ const props = defineProps<{
 	type?: InputTypeHTMLAttribute | 'nist';
 }>();
 
-const emit = defineEmits(['update:modelValue']);
+const emit = defineEmits(['update:modelValue', 'onFocusOut']);
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/widgets/tera-input.vue:32]:
```suggestion
const emit = defineEmits(['update:modelValue', 'on-focus-out']);
```

**Review by shawnyama — APPROVED** [review]
lgtm

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts [hunk 2]
```diff
@@ -51,8 +51,10 @@ export interface OptimizeCiemssOperationState extends BaseState {
 	// Charts + Outputs:
 	chartConfigs: string[][];
 	inProgressOptimizeId: string;
-	inProgressForecastId: string;
-	forecastRunId: string;
+	inProgressPreForecastId: string;
+	preForecastRunId: string;
+	inProgressPostForecastId: string;
+	postForecastRunId: string;
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:57]: Your choice of names may be a bit confusing? Isn't it more like [pre-optimize, post-optimize], instead of [pre-forecast, post-forecast]?

File: packages/client/hmi-client/src/components/widgets/tera-input.vue [hunk 3]
```diff
@@ -82,7 +83,7 @@ onMounted(() => {
 const unmask = () => {
 	// convert back to a number when finished
 	if (isNistType && !getErrorMessage.value) {
-		emit('update:modelValue', nistToNumber(maskedValue.value));
+		emit('update:model-value', nistToNumber(maskedValue.value));
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/widgets/tera-input.vue:86]: Do we need to change all other places that use tera-input?

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/widgets/tera-input.vue:86]: Others were already referring to this as `model-value` I checked!

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts (hunk 2 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:57]: hmm yes but they are both IDs for forecasts.
[pre-optimized-forecast] is a bit wordy.
Maybe firstForecast, and secondForecast? Im not certain but am open to suggestions!

Comment (mwdchang) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:57]: Oh I think I am reading them as:

```
preForecast-runId
postForecast-runId
```
but you are implying
```
pre-forecastRunId
post-forecastRunId
```

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:57]: Correct

**Review by mwdchang — APPROVED** [review]
